product_listing.css

```diff
@@ -0,0 +1,200 @@
+* {
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+  --primary-color: #03045e;
+  --text-color: white;
+  --black-color: black;
+  --hover-btn-color: #caf0f8;
+}
+
+body {
+  font-family: sans-serif;
+}
+
+#page {
+  display: grid;
+  width: 100vw;
+  height: 100vh;
+  grid-template-areas:
+    "head head"
+    "filter content"
+    "filter content"
+    "footer footer";
+  grid-template-columns: 20% 80%;
+  grid-template-rows: 5rem 50% 50% 10rem;
+  grid-gap: 0;
+}
+
+#page header {
+  grid-area: head;
+}
+
+#page footer {
+  grid-area: footer;
+}
+#page sidebar {
+  background-color: var(--text-color);
+  grid-area: filter;
+  position: -webkit-sticky;
+  position: sticky;
+  top: 20px;
+}
+
+@media only screen and (max-width: 600px) {
+  #page {
+    grid-template-areas:
+      "head"
+      "filter"
+      "content"
+      "footer";
+    grid-template-columns: 1fr;
+  }
+}
+
+.header-row {
+  grid-area: head;
+  background-color: var(--primary-color);
+  justify-content: space-around;
+  align-items: center;
+  flex-wrap: wrap;
+  color: var(--text-color);
+  height: 5rem;
+}
+
+.header-row .nav-items button {
+  background-color: var(--text-color);
+  color: var(--black-color);
+}
+
+.header-row .nav-items .number-red {
+  background-color: tomato;
```

Comment:
Try to use variable for color here as well as you are using variables everywhere.